site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx

```diff
@@ -279,6 +297,31 @@ const useExternalAuthVerification = (
   };
 };
 
+function DuplicateWarningMessage() {
+  const [isDismissed, dismiss] = useReducer(() => true, false);
```

Comment:
this is clever, but maybe too clever. 😅 it took me a second to realize what this does, and others who are less familiar with `useReducer` could probably be quite confused by this. can we just use `useState` instead?

Reply:
Yeah, that's fair, and I figured I'd get some pushback on that.

I use the action-less reducer pattern a lot in my personal code, where I want to make my expected state transitions super, super explicit (in this case, making it clear that state can only ever transition from false to true, and can't be undone). It helps that useReducer has function overloads to augment the type of the dispatch if it is action-less

But as far as readability/communication, it's probably a bit much/too convoluted for a simple use case like this